### PR TITLE
[openssl-unix] Add mingw with dynamic libs support - take 2

### PR DESCRIPTION
**Pull request by longnguyen2004**
This is a rework of my previous PR, hopefully without any flaws

### CODE REVIEW

File: ports/openssl-unix/CONTROL [hunk 1]
```diff
@@ -1,4 +1,4 @@
 Source: openssl-unix
-Version: 1.1.1g
+Version: 1.1.1g-1
```
Comment (NancyLi1013) [inline ports/openssl-unix/CONTROL:2]:
```suggestion
Version: 1.1.1g
Port-Version: 1
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/openssl-unix/CONTROL (hunk 1 above)
Comment (longnguyen2004) [reply ports/openssl-unix/CONTROL]: Done

**Review by JackBoosY — APPROVED** [review]

File: ports/openssl-unix/CMakeLists.txt [hunk 2]
```diff
@@ -76,10 +82,15 @@ if(BUILD_SHARED_LIBS)
     if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
         set(LIB_EXT dylib)
         set(LIB_EXTS ${SHLIB_VERSION}.${LIB_EXT})
+	elseif(MINGW)
```
Comment (strega-nil) [inline ports/openssl-unix/CMakeLists.txt:85]: Could you reindent this?

File: ports/openssl-unix/CMakeLists.txt [hunk 3]
```diff
@@ -76,10 +82,15 @@ if(BUILD_SHARED_LIBS)
...
         set(LIB_EXTS ${SHLIB_VERSION}.${LIB_EXT})
+	elseif(MINGW)
+		string(REPLACE "." "_" SHLIB_VERSION "${SHLIB_VERSION}")
+		set(BIN_EXT dll)
+		set(LIB_EXT dll.a)
     else()
         set(LIB_EXT so)
         set(LIB_EXTS ${LIB_EXT}.${SHLIB_VERSION})
     endif()
+	list(APPEND BIN_EXTS ${BIN_EXT})
```
Comment (strega-nil) [inline ports/openssl-unix/CMakeLists.txt:93]: Same here, reindent

File: ports/openssl-unix/portfile.cmake [hunk 4]
```diff
@@ -1,4 +1,6 @@
+if (NOT VCPKG_TARGET_IS_MINGW)
 vcpkg_fail_port_install(MESSAGE "${PORT} is only for openssl on Unix-like systems" ON_TARGET "UWP" "Windows")
```
Comment (strega-nil) [inline ports/openssl-unix/portfile.cmake:2]: Does the target get set to windows on mingw targets? Also, reindent.

**Review by strega-nil — CHANGES_REQUESTED** [review]

File: ports/openssl-unix/portfile.cmake (hunk 4 above)
Comment (longnguyen2004) [reply ports/openssl-unix/portfile.cmake:2]: Yes, it sets both `VCPKG_TARGET_IS_WINDOWS` and `VCPKG_TARGET_IS_MINGW`

**Comment by longnguyen2004** [comment]
I'll do the reindenting when I got home, currently on vacation :D

**Comment by strega-nil** [comment]
Thanks so much @longnguyen2004 :D

File: ports/openssl-unix/CMakeLists.txt (hunk 2 above)
Comment (longnguyen2004) [reply ports/openssl-unix/CMakeLists.txt]: Done

File: ports/openssl-unix/CMakeLists.txt (hunk 3 above)
Comment (longnguyen2004) [reply ports/openssl-unix/CMakeLists.txt]: Done

**Pull request merged**